### ISSUE TITLE
Black levels issue on Quest 1 OLED display

### ISSUE THREAD

**Issue by fakedeltatime** [alvr-org/ALVR#2434]
## Description
The colors, especially black, on the Oculus Quest 1 seem to be somewhat off, with the biggest issue being that black is gray. This happens in game environments, as well as when the stream is not enabled but the streamer is connected (when having the black screen issue for example).
![alvr_black_levels](https://github.com/user-attachments/assets/fb8faf6a-3a05-4e5a-8e8b-3548868ff615)

## General Troubleshooting
- [x] I carefully followed the instructions in the [README](https://github.com/alvr-org/ALVR/blob/master/README.md) and successfully completed the setup wizard
- [x] I read the [ALVR GitHub Wiki](https://github.com/alvr-org/ALVR/wiki)

### Hardware
[`hw-probe`](https://linux-hardware.org/?probe=a45b22c07c)

### Installation
**ALVR Version**: 20.6.1

**ALVR Settings File**:?

**SteamVR Version**: 2.7.4

**Install Type**:
- [x] Packaged (`exe`, `deb`, `rpm`, etc)
- [ ] Portable (`zip`)
- [ ] Source

**OS Name and Version**: (Linux) NixOS 23.11 (Tapir)

**Comment by The-personified-devil** [comment]
Alvr has color settings, you might want to adjust those

**Comment by fakedeltatime** [comment]
> Alvr has color settings, you might want to adjust those

I forgot to mention in the original post that I did give the color correction settings a try, but to no avail. 

Even setting brightness to zero, which also resulted in a (visually) completely _black_ display, rendered a shade of gray in the headset. The ALVR native blank environment with connection instructions has properly black elements.

**Comment by The-personified-devil** [comment]
> rendered a shade of gray in the headset

That's probably on the display then, it can't properly present black

> The ALVR native blank environment with connection instructions has properly black elements

Could you explain this further

**Comment by fakedeltatime** [comment]
> That's probably on the display then, it can't properly present black

No, the display can present black properly (with clearly turned off pixels), since it is OLED, as it does in standalone content (such as the ALVR headset client)

> Could you explain this further

I mean that the dark grid environment that the headset client of ALVR opens has perfect blacks, where as the stream does not. Going from the native black levels to the non-black blacks of the streamed content is jarring.

**Comment by CactusBoy3** [comment]
Can confirm I have the same issue. Running KDE neon, in Half Life Alyx, around chapter 3 ish where it is dark, it looks like I am using an LCD headset. Very grey rather than black.

**Comment by treirouxel88** [comment]
I was thinking that there was a problem since I went from ALVR 20.10.0 -> 20.11.1, the colors appear lighter, my impression in games is as if I launched Discord in light theme...

**Comment by jadealchemy** [comment]
Is it possible to reopen this given it is still an issue.
Ubuntu 25.10
AMD RX 9060 XT 16GB
ALVR 20.14.1

Also tested with ALVR 20.10.0 and v21.0.0-dev12+nightly.2025.12.10 without any change.
Adjusting post process does not help me.

**Comment by jadealchemy** [comment]
Continuing to mess with this I found that disabling "Full range color" and enabling "Override for full range color" fixes the issue.

If your like me and found this issue on google. The buttons are under Settings, Video, expand "Encoder config".

**Comment by jadealchemy** [comment]
My assumption is that the encoder is not honouring the full range colour setting but the headset still clamps the range as if it was receiving a full range stream. Testing with a Quest 2 seems to confirm this is not a Quest 1 exclusive issue.

**Comment by S0yKaf** [comment]
>disabling "Full range color" and enabling "Override for full range color"

This fixed it for me on my quest 2
